qiskit/visualization/pulse_visualization.py

```diff
@@ -15,96 +15,67 @@
 # pylint: disable=invalid-name
 
 """
-Visualization function for pulse envelope.
+matplotlib pulse visualization.
 """
 
-import numpy as np
-from scipy.interpolate import CubicSpline
+import logging
 
-from qiskit.exceptions import QiskitError
+from qiskit.pulse import Schedule, Instruction, SamplePulse
+from qiskit.visualization.exceptions import VisualizationError
+from qiskit.visualization import matplotlib as _matplotlib
 
+logger = logging.getLogger(__name__)
 
-def pulse_drawer(samples, duration, dt=None, interp_method='None',
-                 filename=None, interactive=False,
-                 dpi=150, nop=1000, size=(6, 5)):
+
+def pulse_drawer(data, dt=1, style=None, filename=None, output='mpl',
+                 interp_method=None, scaling=None, channels_to_plot=None,
+                 plot_all=False, plot_range=None, interactive=False,
+                 table=True, label=False, framechange=True):
     """Plot the interpolated envelope of pulse
 
     Args:
-        samples (ndarray): Data points of complex pulse envelope.
-        duration (int): Pulse length (number of points).
-        dt (float): Time interval of samples.
-        interp_method (str): Method of interpolation
-            (set `None` for turn off the interpolation).
-        filename (str): Name required to save pulse image.
+        data (ScheduleComponent or SamplePulse): Data to plot
+        dt (float): Time interval of samples
+        style (OPStylePulse or OPStyleSched): A style sheet to configure plot appearance
+        filename (str): Name required to save pulse image
+        output (str): Select the output method to use for drawing the pulse object.
+            Valid choices are `mpl`.
+        interp_method (Callable): interpolation function
+            See `qiskit.visualization.interpolation` for more information
+        scaling (float): scaling of waveform amplitude
```

Comment:
Aren't all pulse amplitudes in [0,1]?

Reply:
Should be. I'll need to look into what this does.